lib/shared/bucketing-assembly-script/assembly/helpers/jsonHelpers.ts

```diff
@@ -38,23 +38,43 @@ export function getStringFromJSONOptional(jsonObj: JSON.Obj, key: string): strin
     }
 }
 
-export function isValidString(jsonObj: JSON.Obj, key: string, validStrings: string[]): string {
+export function isValidString(
+    jsonObj: JSON.Obj,
+    key: string,
+    validStrings: string[],
+    shouldThrow: bool = true
+): string {
     const value = getStringFromJSON(jsonObj, key)
     if (!validStrings.includes(value)) {
-        throw new Error(`Invalid string value: ${value}, for key: ${key}, must be one of: ${validStrings.join(', ')}`)
+        if (shouldThrow) {
+            throw new Error(
+                `Invalid string value: ${value}, for key: ${key}, must be one of: ${validStrings.join(', ')}`
+            )
+        } else {
+            console.warn(`[DevCycle] Warning: String value: ${value}, for key: ${key} does not match a valid string.`)
```

Comment:
We should use log here. Warn will sigkill.

Reply:
er - more accurately - warn will do different things per impl.

Reply:
why would it do that?

Reply:
depends on the WASI level of implementation. All these calls are bindings to WASI.

Reply:
are the implementations of warn any more divergent than `log`?

Reply:
My understanding is that warn is generally lumped in with error for most implementations that don't have full WASI bindings (ie PHP) - but for languages with full WASI support (C#, Go, Node) it's fine and can be handled separately

Reply:
going to revert back to just `console.log` then